tests/test_permissions.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+import responses
+
+from pyartifactory import ArtfictoryPermission
+from pyartifactory.exception import (
+    PermissionNotFoundException,
+    PermissionAlreadyExistsException,
+)
+from pyartifactory.models.Auth import AuthModel
+from pyartifactory.models.Permission import Permission, SimplePermission
+
+URL = "http://localhost:8080/artifactory"
+AUTH = ("user", "password_or_apiKey")
+
+SIMPLE_PERMISSION = SimplePermission(name="test_permission", uri="someuri")
+PERMISSION = Permission(
+    **{
+        "name": "test_permission",
+        "repositories": ["test_repository"],
+        "principals": {
+            "users": {"test_user": ["r", "w", "n", "d"]},
+            "groups": {"developers": ["r"]},
+        },
+    }
+)
+
+
+class TestGroup:
+    @staticmethod
+    @responses.activate
+    def test_create_permission_fail_if_group_already_exists(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=200,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "get")
+        with pytest.raises(PermissionAlreadyExistsException):
+            artifactory_permission.create(PERMISSION)
+
+        artifactory_permission.get.assert_called_once_with(PERMISSION.name)
+
+    @staticmethod
+    @responses.activate
+    def test_create_permission_success(mocker):
+        responses.add(
+            responses.GET,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=404,
+        )
+        responses.add(
+            responses.PUT,
+            f"{URL}/api/security/permissions/{PERMISSION.name}",
+            json=PERMISSION.dict(),
+            status=201,
+        )
+
+        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        mocker.spy(artifactory_permission, "get")
+        with pytest.raises(PermissionNotFoundException):

```

Comment:
Is this supposed to raise? The creation succeeds, I don't expect a "Permission not found".

Reply:
Yes, it's normal. The way we implemeted it is: If the permission does not exist, then we create it